internal/service/streamprocessor/model_test.go

```diff
@@ -90,21 +97,36 @@ func streamProcessorWithStats(t *testing.T) *admin.StreamsProcessorWithStats {
 		t.Fatal(err)
 	}
 	processor.SetStats(stats)
+	if options != nil {
+		processor.SetOptions(*options)
+	}
 	return processor
 }
 
-func streamProcessorDSTFModel(t *testing.T, state, stats string) *streamprocessor.TFStreamProcessorDSModel {
+func streamProcessorDSTFModel(t *testing.T, state, stats string, options types.Object) *streamprocessor.TFStreamProcessorDSModel {
 	t.Helper()
 	return &streamprocessor.TFStreamProcessorDSModel{
 		ID:            types.StringValue(processorID),
 		InstanceName:  types.StringValue(instanceName),
+		Options:       options,
 		Pipeline:      types.StringValue("[{\"$source\":{\"connectionName\":\"sample_stream_solar\"}},{\"$emit\":{\"connectionName\":\"__testLog\"}}]"),
 		ProcessorName: types.StringValue(processorName),
 		ProjectID:     types.StringValue(projectID),
 		State:         conversion.StringNullIfEmpty(state),
 		Stats:         conversion.StringNullIfEmpty(stats),
 	}
 }
+
+func optionsToTFModel(t *testing.T, options *admin.StreamsOptions) types.Object {
+	t.Helper()
+	ctx := context.Background()
+	result, diags := streamprocessor.ConvertOptionsToTF(ctx, options)
+	if diags.HasError() {
+		t.Fatal(diags)
+	}
+	return *result
```

Comment:
can't result ever be nil here?

Reply:
No, either diags.HasError() and t.Fatal() exits or result is set.
But could add `assert.NotNil()` just to be sure

Reply:
oh, it's in a test, it's not so important then.  yeah, NotNil can be useful

Reply:
Updated in 908af3b0cce669da6aaa696144701bf29e16059a